### PR TITLE
Added test for testing sciptpubkey of closing message per bolt 2 spec

### PR DESCRIPTION
**Pull request by SWvheerden**
I added a test to check if the sciptpubkey is either one of the 4 required scriptbupkey's. 

I flagged another test, I am not sure if that is not required. I will update the code pending the review.

### CODE REVIEW

**Comment by TheBlueMatt** [comment]
I'd actually kinda like to figure out *why* the BOLT suggests that we check it against an acceptable type. Like, checking the size is definitely useful for fee reasons, but if its within the right size, why should we care what script they want to use?

File: src/ln/channel.rs [hunk 1]
```diff
@@ -2098,7 +2102,7 @@ impl Channel {
 		if !self.pending_inbound_htlcs.is_empty() || !self.pending_outbound_htlcs.is_empty() {
 			return Err(HandleError{err: "Remote end sent us a closing_signed while there were still pending HTLCs", action: None});
 		}
-		if msg.fee_satoshis > 21000000 * 10000000 {
+		if msg.fee_satoshis > 21000000 * 10000000 { //is this test required? We check for a much lower amount below, thus if that one will fail, this one will always fail?
```
[the remark this answers is not in the document]
Comment (SWvheerden) [reply src/ln/channel.rs]: ok cwl, wasnt exactly sure, I will leave a comment about this.

**Comment by SWvheerden** [comment]
This is the reason they give in the official bolt specs:
The scriptpubkey forms include only standard forms accepted by the Bitcoin network, which ensures the resulting transaction will propagate to miners
Which I guess is valid as you want to make sure the transaction goes through.

File: src/ln/channel.rs [hunk 2]
```diff
@@ -2012,7 +2012,11 @@ impl Channel {
 		if self.channel_outbound && msg.scriptpubkey.len() > 34 {
 			return Err(HandleError{err: "Got shutdown_scriptpubkey of absurd length from remote peer", action: None});
 		}
-		//TODO: Check shutdown_scriptpubkey form as BOLT says we must? WHYYY
+
+		//Check shutdown_scriptpubkey form as BOLT says we must
+		if !(msg.scriptpubkey.is_p2pkh())&&!(msg.scriptpubkey.is_p2sh())&&!(msg.scriptpubkey.is_v0_p2wpkh())&&!(msg.scriptpubkey.is_v0_p2wsh()){
```
Comment (ariard) [inline src/ln/channel.rs]: nit: adding a space before/after AND operator isn't nicer for code readability ?

File: src/ln/channel.rs [hunk 3]
```diff
@@ -2012,7 +2012,12 @@ impl Channel {
 		if self.channel_outbound && msg.scriptpubkey.len() > 34 {
 			return Err(HandleError{err: "Got shutdown_scriptpubkey of absurd length from remote peer", action: None});
 		}
-		//TODO: Check shutdown_scriptpubkey form as BOLT says we must? WHYYY
+
+		//Check shutdown_scriptpubkey form as BOLT says we must
+		if !(msg.scriptpubkey.is_p2pkh()) && !(msg.scriptpubkey.is_p2sh())
+			&& !(msg.scriptpubkey.is_v0_p2wpkh()) && !(msg.scriptpubkey.is_v0_p2wsh()){
```
[the remark this answers is not in the document]
Comment (yuntai) [reply src/ln/channel.rs]: nit: redundant parentheses 
https://doc.rust-lang.org/reference/expressions.html#expression-precedence

**Pull request merged**